crates/bevy_dev_tools/src/fps_overlay.rs

```diff
@@ -0,0 +1,97 @@
+//! Module containing logic for FPS overlay.
+
+use bevy_app::{Plugin, Startup, Update};
+use bevy_asset::Handle;
+use bevy_color::Color;
+use bevy_diagnostic::{DiagnosticsStore, FrameTimeDiagnosticsPlugin};
+use bevy_ecs::{
+    component::Component,
+    query::With,
+    schedule::{common_conditions::resource_changed, IntoSystemConfigs},
+    system::{Commands, Query, Res, Resource},
+};
+use bevy_text::{Font, Text, TextSection, TextStyle};
+use bevy_ui::node_bundles::TextBundle;
+
+/// A plugin that adds an FPS overlay to the Bevy application.
+///
+/// This plugin will add the [`FrameTimeDiagnosticsPlugin`] if it wasn't added before.
+///
+/// Note: It is recommended to use native overlay of rendering statistics when possible for lower overhead and more accurate results.
+/// The correct way to do this will vary by platform:
+/// - **Metal**: setting env variable `MTL_HUD_ENABLED=1`
+#[derive(Default)]
+pub struct FpsOverlayPlugin {
+    /// Starting configuration of overlay, this can be later be changed through [`FpsOverlayConfig`] resource.
+    pub config: FpsOverlayConfig,
+}
+
+impl Plugin for FpsOverlayPlugin {
+    fn build(&self, app: &mut bevy_app::App) {
+        if !app.is_plugin_added::<FrameTimeDiagnosticsPlugin>() {
```

Comment:
It might be worth adding a note here to change this line when plugin dependencies gets implemented.